modelica_builder/model.py

```diff
@@ -116,6 +121,18 @@ def edit_connect(self, port_a, port_b, new_port_a=None, new_port_b=None):
                         .chain(NthChildSelector(4)))
             self.add(SimpleTransformation(selector, Edit.make_replace(new_port_b)))
 
+    def insert_equation_for_loop(self, index_identifier, expression_raw, loop_body_raw_list):
+        """constructs and inserts a for loop in the equation section
+
+        :param index_identifier: string, identifier for the index variable
+        :param expression_raw: string, the expression for iteration, e.g. range(x)
+        :param loop_body_raw: list, list of loop body lines, each of which is valid modelica code
+        """
+        self.add(
+            EquationForLoopBuilder(
+                index_identifier, expression_raw, loop_body_raw_list
+            ).transformation())
+
```

Comment:
Design note: decided to make the body a list rather than a single string as it'd allow us to apply our own indentation line by line, rather than making the user to do it themselves